packages/backend-auth/src/userpool_access_policy_factory.ts

```diff
@@ -22,127 +14,99 @@ export class UserPoolAccessPolicyFactory {
   private policyCount = 1;
 
   /**
-   * Instantiate with the bucket to generate policies for
+   * Instantiate with the userpool to generate policies for
    */
   constructor(private readonly userpool: IUserPool) {
     this.stack = Stack.of(userpool);
   }
 
-  createPolicy = (actions: Readonly<Set<UserpoolAction>>) => {
-    if (actions.size === 0) {
+  createPolicy = (actions: AuthActions) => {
+    const policyActions: Set<string> = new Set();
+
+    actions.forEach((authAction) => {
+      const mappedAction = iamActionMap[authAction];
+
+      if (typeof mappedAction === 'string') {
+        policyActions.add(mappedAction);
+      } else {
+        mappedAction.forEach((action) => policyActions.add(action));
+      }
+    });
+
+    if (policyActions.size === 0) {
       throw new AmplifyFault('EmptyPolicyFault', {
         message: 'At least one action must be specified',
       });
     }
 
-    const statements: PolicyStatement[] = [];
-
-    actions.forEach((action) => {
-      statements.push(this.getStatement(action));
-    });
+    const policy = new Policy(
+      this.stack,
+      `${this.namePrefix}${this.policyCount++}`,
+      {
+        statements: [
+          new PolicyStatement({
+            actions: [...policyActions],
```

Comment:
```suggestion
            actions: policyActions,
```

Reply:
policyActions is a Set, actions is expecting string[]